### PR TITLE
mosquitto: remove build time to fix reproducible builds

### PR DESCRIPTION
**Pull request by lynxis**
Maintainer: @karlp
Compile tested: lantiq

Build timestamps prevents reproducible builds [0].

[0] https://reproducible-builds.org/docs/timestamps/

Signed-off-by: Alexander Couzens <lynxis@fe80.eu>

### CODE REVIEW

**Comment by champtar** [comment]
@karlp @lynxis seems there is a dependency issue

File: net/mosquitto/patches/100-reproducible-builds.patch [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
++	_mosquitto_log_printf(NULL, MOSQ_LOG_INFO, "mosquitto version %s starting", VERSION);
+ 	if(config.config_file){
+ 		_mosquitto_log_printf(NULL, MOSQ_LOG_INFO, "Config loaded from %s.", config.config_file);
+ 	}else{
+@@ -308,8 +308,8 @@ int main(int argc, char *argv[])
+ 		/* Set static $SYS messages */
+ 		snprintf(buf, 1024, "mosquitto version %s", VERSION);
+ 		mqtt3_db_messages_easy_queue(&int_db, NULL, "$SYS/broker/version", 2, strlen(buf), buf, 1);
+-		snprintf(buf, 1024, "%s", TIMESTAMP);
++		snprintf(buf, 1024, "%s", "1970-01-01 00:00:00+0000");
```
Comment (karlp) [inline net/mosquitto/patches/100-reproducible-builds.patch]: I thought you normally put in a string like "REPRODUCIBLE BUILD" not a string that looks like a broken clock?

**Comment by karlp** [comment]
Have you filed this upstream?  I can work there on this.

**Comment by champtar** [comment]
```
Package libmosquittopp is missing dependencies for the following libraries:
libmosquitto.so.1
```

**Comment by karlp** [comment]
That libmosquittopp issue is separate, I'll try and look at that, but it might be monday.

**Comment by champtar** [comment]
@karlp I know, just pointing it out so it's not forgotten ;)

**Comment by lynxis** [comment]
@karlp no. i'ven't yet filled this upstream.

**Comment by champtar** [comment]
maybe related #4751

**Comment by karlp** [comment]
Yep same bug, they just never got back to me with testing it.

**Comment by lynxis** [comment]
https://github.com/eclipse/mosquitto/pull/651

**Comment by karlp** [comment]
This pull still has the older form patch

**Comment by karlp** [comment]
merged in 0bc06df68cf6d0154e7f76edd3935c500c9a4510 thanks!

**Pull request closed without merging**